src/apps/config/leader.lua

```diff
@@ -70,16 +68,25 @@ end
 
 function Leader:set_initial_configuration (configuration)
    self.current_configuration = configuration
-   self.current_app_graph = self.setup_fn(configuration)
    self.current_in_place_dependencies = {}
-   self.current_in_place_dependencies =
-      self.support.update_mutable_objects_embedded_in_app_initargs (
-         {}, self.current_app_graph, self.schema_name, 'load', '/',
-         self.current_configuration)
-   local initial_app_graph = app_graph.new() -- Empty.
-   local actions = self.support.compute_config_actions(
-      initial_app_graph, self.current_app_graph, {}, 'load')
-   self:enqueue_config_actions(actions)
+
+   -- Start the followers and configure them.
+   local follower_configurations = self.setup_fn(configuration)
+
+   for id, config in pairs(follower_configurations) do
+      local follower_pid = self.worker_start_fn()
+      local in_place_dependencies =
+         self.support.update_mutable_objects_embedded_in_app_initargs (
+            app_graph.new(), config, self.schema_name, 'load', '/',
+            self.current_configuration)
```

Comment:
Hum!  Here I'm not sure.  So the leader is exposing a single YANG tree, and then the setup function maps that onto N app graphs.  But when a user updates a part of that YANG tree, some subset of those N graphs will have been updated in place.  I think we need to build one "in_place_dependencies" for the whole system so we can know what to restart when a user updates the YANG configuration -- i.e. instead of having update_mutable_objects_embedded_in_app_initargs take a single app graph, probably it should take a table of all app graphs, keyed by follower ID.  WDYT?